src/main/java/slimeknights/tconstruct/library/recipe/casting/ItemCastingRecipe.java

```diff
@@ -27,21 +28,27 @@
   /** 'cast' item for recipe (doesn't have to be an actual 'cast') */
   @Getter
   protected final Ingredient cast;
-  /** FluidStack for recipe (fluid and amount) */
-  @Getter
+  @Getter // For JEI
   protected final FluidIngredient fluid;
   protected final ItemStack result;
-  /** How long it takes to cool off */
   @Getter
   protected final int coolingTime;
-  /** Whether the cast is consumed when cast */
   @Getter
   protected final boolean consumed;
   protected final boolean switchSlots;
 
   @Override
-  public ItemStack getCraftingResult(ICastingInventory inv) {
-    return this.result.copy();
+  public boolean matches(ICastingInventory inv, World worldIn) {
+    return this.getCast().test(inv.getStack()) && this.getFluid(inv).test(inv.getFluid());
+  }
+
+  public FluidIngredient getFluid(ICastingInventory inv) {
```

Comment:
What is the reason for this method? Its not in the interface, so we might as well in matches use `this.fluid`